src/main/java/com/vertispan/webdriver/gwt/gwtdriver/by/ByWidgetChildren.java

```diff
@@ -0,0 +1,132 @@
+   * etc, as these are all subclasses of {@code ValueBoxBase}.  Note that interfaces cannot be used,
+   * only base classes, and those classes *must* extend Widget.
+   *
+   * @param widgetType The type of widget to find
+   */
+  public ByWidgetChildren(Class<? extends Widget> widgetType) {
+    this(widgetType.getName());
+  }
+
+  /**
+   * Finds the descendant Widgets of the given type.
+   * <p>
+   * This will find any subtype of that widget, allowing you to pass in {@link
+   * com.google.gwt.user.client.ui.ValueBoxBase} and find any {@link com.google.gwt.user.client.ui.TextBox},
+   * {@link com.google.gwt.user.client.ui.TextArea}, {@link com.google.gwt.user.client.ui.IntegerBox},
+   * etc, as these are all subclasses of {@code ValueBoxBase}.  Note that interfaces cannot be used,
+   * only base classes, and those classes *must* extend Widget.
+   *
+   * @param driver The driver to use to communicate with the browser.
+   * @param widgetType The type of widget to find
+   */
+  public ByWidgetChildren(WebDriver driver, Class<? extends Widget> widgetType) {
+    this(driver, widgetType.getName());
+  }
+
+  /**
+   * Finds the descendant Widgets of the given type.
+   * <p>
+   * This will find any subtype of that widget, allowing you to pass in {@link
+   * com.google.gwt.user.client.ui.ValueBoxBase} and find any {@link com.google.gwt.user.client.ui.TextBox},
+   * {@link com.google.gwt.user.client.ui.TextArea}, {@link com.google.gwt.user.client.ui.IntegerBox},
+   * etc, as these are all subclasses of {@code ValueBoxBase}.  Note that interfaces cannot be used,
+   * only base classes, and those classes *must* extend Widget.
+   *
+   * @param widgetClassName The type of widget to find
+   */
+  public ByWidgetChildren(String widgetClassName) {
+    this(null, widgetClassName);
+  }
+
+  /**
+   * Finds the descendant Widgets of the given type.
+   * <p>
+   * This will find any subtype of that widget, allowing you to pass in {@link
+   * com.google.gwt.user.client.ui.ValueBoxBase} and find any {@link com.google.gwt.user.client.ui.TextBox},
+   * {@link com.google.gwt.user.client.ui.TextArea}, {@link com.google.gwt.user.client.ui.IntegerBox},
+   * etc, as these are all subclasses of {@code ValueBoxBase}.  Note that interfaces cannot be used,
+   * only base classes, and those classes *must* extend Widget.
+   *
+   * @param driver The driver to use to communicate with the browser.
+   * @param widgetClassName The type of widget to find
+   */
+  public ByWidgetChildren(WebDriver driver, String widgetClassName) {
+    super(driver);
+    this.widgetClassName = widgetClassName;
+  }
+
+  @Override
+  public List<WebElement> findElements(SearchContext context) {
+    Require.nonNull("Search Context", context);
+    final WebElement contextElem = toWebElement(context);
+    ExportedMethods m = ClientMethodsFactory.create(ExportedMethods.class, getDriver(context));
```

Comment:
I wonder if we shouldn't ThreadLocal/WeakRef map these instances to avoid re-creating them each call, and make them easy to look up on the current driver instance...

Reply:
maybe.  future update. I'll add new ticket

Reply:
#11